tracker/interface.go

```diff
@@ -70,6 +70,11 @@ type Interface interface {
 	// OnChanged is a callback to register with the InformerFactory
 	// so that we are notified for appropriate object changes.
 	OnChanged(obj interface{})
+
+	// OnDeletedParent is a callback to register with the InformerFactory
+	// so that we are notified for deletions of a watching parent to
+	// remove the respective tracking.
+	OnDeletedParent(obj interface{})
```

Comment:
Going to paint the shed!

To me "Parent" evokes a controller relationship (e.g. this resource produced the resources being watched), which may be the case, but isn't always (e.g. Route -> Config/Rev, or ObjectRefs).

I think the name I'd use is probably `Observer` to avoid the implied hierarchical relationship.  WDYT?

Reply:
SG!